packages/core/src/EntityManager.ts

```diff
@@ -69,6 +86,16 @@ export class EntityManager<D extends IDatabaseDriver = IDatabaseDriver> {
   private readonly entityFactory: EntityFactory = new EntityFactory(this);
   private readonly unitOfWork: UnitOfWork = new UnitOfWork(this);
   private readonly resultCache = this.config.getResultCacheAdapter();
+  private readonly refLoader = new DataLoader<Ref<any>, any>(async refs => {
+    const groupedIds = groupPrimaryKeysByEntity(refs);
+    const promises = Array.from(groupedIds).map(
+      async ([entity, ids]) =>
+        await this.getRepository(entity).find(Array.from(ids)),
```

Comment:
don't use repositories internally
```suggestion
        await this.find(entity, Array.from(ids)),
```

also the first parameter is rather `entityName`, calling this just `entity` is pretty confusing. and i dont think we need to mark the method as async and await the inside, just pass the promise already created by `em.find`.

Reply:
No we don't need async/await, it's just [linting](https://typescript-eslint.io/rules/promise-function-async/) from my existing dataloader library which uses `standard-with-typescript`.